### PR TITLE
docs: Set up docs site infra and refactor existing docs into the site.

### PR DESCRIPTION
**Pull request by junhaoliao**
# Description

This PR sets up docs site infra as done in https://github.com/y-scope/clp/tree/main/docs and https://github.com/y-scope/clp-ffi-py/tree/main/docs .

## TODO
1. Add GH workflow to auto publish the docs.
2. Update https://github.com/y-scope/yscope-docs to include this project. Issue tracked at https://github.com/y-scope/yscope-docs/issues/22
3. Add contents in user guide.

# Validation performed

1. Ran `task docs:serve` and observed the debug server started which serves the static site locally at http://127.0.0.1:8080 .
4. Inspected the docs for accuracy and to avoid grammar errors.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208827675629273
  - https://app.asana.com/0/0/1208789844606457

## Summary by CodeRabbit

- **New Features**
	- Added a new submodule for external utilities to enhance project functionality.
	- Introduced a structured task management system for building and serving documentation.
	- New sections added to the developer guide, including "Getting Started" and "Design Overview."
	- Added a script to check for broken documentation links.
	- Introduced a new CSS file for improved theme styling.
	- Enhanced support for compressed log files and JSON log files in the YScope Log Viewer.

- **Documentation**
	- Updated README for clearer bug reporting and feature request instructions.
	- Introduced a user guide for operating the YScope Log Viewer.
	- Added detailed setup instructions for building and running the project.
	- Enhanced documentation on validation and linting processes.

- **Bug Fixes**
	- Improved handling of PNG files in Git LFS for better storage management.

- **Chores**
	- Updated dependencies in documentation requirements for improved build processes.

### CODE REVIEW

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: docs/src/index.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Docs for those interested in developing and contributing to the YScope Log Viewer.
+:::
+::::
+
+# Online demo
+
+A demo of the viewer is available [here][online-demo].
+
+* The demo loads a Hadoop YARN log file from the [hive-24hrs] log dataset.
+  * More info on the dataset and other datasets can be found [here][datasets].
```
[the remark this answers is not in the document]
Comment (junhaoliao) [reply docs/src/index.md:40]: @kirkrodrigues What the rabbit suggested here is trivial but valid: GET requests to https://zenodo.org/record/7094921 receives
```
HTTP/1.1 301 MOVED PERMANENTLY
...
location: /records/7094921
```
Shall we apply and update also the `README.md`?

[the remark this answers is not in the document]
Comment (kirkrodrigues) [reply docs/src/index.md:40]: Sg

File: docs/src/.gitattributes
[the remark this answers is not in the document]
Comment (kirkrodrigues) [reply docs/src/.gitattributes:1]: Any thoughts?

[the remark this answers is not in the document]
Comment (junhaoliao) [reply docs/src/.gitattributes:1]: > LFS introduces a bit of friction for folks working with the repo.

sure

**Review by kirkrodrigues — APPROVED** [review]
For the PR title, how about:

```
docs: Set up docs site infra and refactor existing docs into the site.
```

**Pull request merged**